src/librustc_codegen_ssa/mir/operand.rs

```diff
@@ -341,24 +341,27 @@ impl<'a, 'tcx, V: CodegenObject> OperandValue<V> {
             .unwrap_or_else(|| bug!("indirect_dest has non-pointer type: {:?}", indirect_dest))
             .ty;
 
-        let (llptr, llextra) = if let OperandValue::Ref(llptr, Some(llextra), _) = self {
+        let (_llptr, llextra) = if let OperandValue::Ref(llptr, Some(llextra), _) = self {
             (llptr, llextra)
         } else {
             bug!("store_unsized called with a sized value")
         };
 
-        // FIXME: choose an appropriate alignment, or use dynamic align somehow
```

Comment:
Looks like back in https://github.com/rust-lang/rust/pull/51131 someone noticed this, but wasn't aware that this is a soundness issue.